internal/exec/stages/files/units.go

```diff
@@ -145,31 +146,41 @@ func parseInstanceUnit(unit types.Unit) (string, string, error) {
 
 // createSystemdPresetFile creates the presetfile for enabled/disabled
 // systemd units.
-func (s *stage) createSystemdPresetFile(presets map[string]*Preset) error {
-	if err := s.relabelPath(filepath.Join(s.DestDir, util.PresetPath)); err != nil {
-		return err
+func (s *stage) createSystemdPresetFiles(presets map[string]*Preset) error {
+
+	//getting directories from presets file for relabling
+	paths := make(map[string]bool)
+	for _, preset := range presets {
+		path := filepath.Dir(s.SystemdPresetPath(preset.scope))
+		if _, value := paths[path]; !value {
+			paths[path] = false
+			if err := s.relabelPath(filepath.Join(s.DestDir, path)); err != nil {
+				return err
+			}
+			paths[path] = true
```

Comment:
We don't use the `false` and `true` values.  `paths` can be a `map[string]struct{}`.

Reply:
`true` or `false` are used here `if _, value := paths[path]; !value {}` to check if paths have been relabelled or not.
This prevent multiple relabel ... that could be in a separate PR as we said previously ... I'm thinking about it

Reply:
Nope, you're ignoring the map value (the first return value) and only checking whether the key is present (the second return value).  So any map value will do, and traditionally `struct{}{}` is used in that case.

Reply:
I will try to manage multiple relabelling in a better way, in another PR, so this piece of code should disappear, it seems to be a better approach isn't it ?